SU-HSDO/suhumsci
# STN-324 : Detail Page Image Styles

# [STN-324](https://sparkbox.atlassian.net/browse/STN-324) READY FOR REVIEW

## Summary
The work in this PR empowers H&S to display images using different images styles when the need arises.

### Acceptance Criteria
* Identify a solution for displaying a non-resized featured-image, within the 3-column-with-image layout, that meets H&S’s needs.
* Ensure that the solution allows for selection of different image styles.
* Implement the approach and demonstrate it’s functionality on the publications detail pages.
* Ensure the approach meets our browser support and accessibility guidelines.

## Need Review By (Date)
4/27

## Urgency
medium

## Steps to Test
1. In the CLI, run `npm run test` and confirm all of our Sass tests pass.
2. Run `npm start` to build the styles. The compiled CSS files were added during a rebase. 🤷  I will rebuild and update them once everything has been reviewed and approved.
2. Review the following pages with the default image style of `Medium Square 360 x 360`
    * You can **check / update the image style** used on a detail page by going to `Structure > Content types > Event _(or whichever node detail page you are working on)_ > Manage Display`. You can also use the contextual link located on the detail page.
    * Click on `Manage layout`
    * Find the `"Image" field` block, click on the contextual link, and select `Configure` 
    * Find `Image Style` and confirm / select `Medium Square (360 x 360)`
    * Update. Save layout. Back to site.
- [ ] Event: http://economics.suhumsci.loc/events/events-all-fields-filled
- [ ] Publication: http://economics.suhumsci.loc/publications/publications-all-fields-filled
- [ ] News: http://economics.suhumsci.loc/news/joeys-test-news-item
- [ ] Person: http://economics.suhumsci.loc/people/ran-abramitzky
3. The work in this PR gives H&S the flexibility to use the image styles set in the configuration. I will be adding recommendations on which image styles to use to the Pattern Doc. Following the steps above, test various image styles: `Medium Scaled (360)`, `Small Square (200 x 200)`, etc.
4. Test another option. 
H&S requested styles for when the image is placed within the main body.
    * Go to http://economics.suhumsci.loc/publications/publications-all-fields-filled
    * Click on the contextual link and select `Manage Display`
    * Click on `Manage layout`
    * Move `"Publications Image"` into the main body section
    * Click on the contextual link for the publications image
    * Add the utility class of `hb-main-body-detail-image` to  the `Field Formatter Class`
    * Update. Save layout. Back to site.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
